lib/etcd.ex

```diff
@@ -75,8 +75,8 @@ defmodule Etcd do
     case Connection.request(srv, :sync, verb, key, query, body, timeout) do
       {:ok,%{"action" => _action}=reply,_} ->
         reply
-      {:ok,%{"errorCode" => errCode, "message" => errMsg},_} ->
-        raise ServerError, code: errCode, message: errMsg
+      {:ok,%{"errorCode" => errCode, "message" => errMsg, "cause" => cause, "index" => index},_} ->
```

Comment:
looks like "cause" field in an error object  is optional (https://github.com/coreos/etcd/blob/master/error/error.go#L116)
so this line may throw a MatchError, can you update it?